src/components/ingestion/HttpIngestionEngine.js

```diff
@@ -0,0 +1,53 @@
+import { PassThrough } from 'stream'
+import fetch from 'node-fetch'
+import AbstractIngestionEngine from '%src/components/ingestion/AbstractIngestionEngine'
+
+/**
+ * The HttpIngestionEngine handles processing a HTTP video stream. It is a concrete
+ * implementation of AbstractIngestionEngine, which reads in a stream,
+ * converts it into an MPEG-TS stream represented as a series of Payloads,
+ * and then pipes that to Kafka.
+ *
+ * @extends AbstractIngestionEngine
+ */
+class HttpIngestionEngine extends AbstractIngestionEngine {
+	// The URL of the stream to read by the engine
+	url = null
+
+	/**
+	 * Create a HttpIngestionEngine.
+	 *
+	 * @param  {string}  url The url of the stream to be ingested
+	 */
+	constructor(url) {
+		if (!url) {
+			throw new Error(
+				'HttpIngestionEngine must be instantiated with a stream URL',
+			)
+		}
+		super()
+
+		this.url = url
+	}
+
+	/**
```

Comment:
Since getInputStream is an override method, rather than replicate the documentation lets use the [`@inheritdoc`](https://jsdoc.app/tags-inheritdoc.html) tag.

NOTE: I think I need to make an issue about this since we're totally not doing that elsewhere.

Reply:
Okay.